compiler/rustc_interface/src/queries.rs

```diff
@@ -347,6 +352,10 @@ impl Compiler {
                 .time("serialize_dep_graph", || gcx.enter(rustc_incremental::save_dep_graph));
         }
 
+        if let Err((path, error)) = queries.finish() {
+            self.session().emit_err(errors::FailedWritingFile { path: &path, error });
+        }
```

Comment:
Would it make sense to make `finish` consume `self` and bubble this down all the way into `DepGraph::finish_encoding()`? We shouldn't touch `GlobalCtxt` after `finish()`. This would require putting it in the context of the timer below, but would allow using `.time()` instead of `.timer()`.

Reply:
I don't know. I need to steal anyway to access the DepGraph, so the move ends up being enforced at runtime anyway.

And I'm not sure how to make the code in rustc_interface work with the higher-ranked lifetime, but I think that's just because I've not used them before.

Reply:
I've had no ideas on how to make this work, so I'm going to drop the commit.